### PR TITLE
[WIP] Allow download/view of shortlink code

### LINKED ISSUE #1640: [REQUEST] Endpoint for "just the text" of a short URL
Requested in slack by Darrell Wright.

```
Is there a way to query the text of CE short url.  Say I wanted to #include <https://gcc.godbolt.org/z/SOMEID>
```

Ideally we'd give an endpoint like the existing RESTful one that "just" yields the editor text. Not sure what it should do if there's more than one editor, pick lowest ID? Error?

### PR DESCRIPTION
**Pull request by partouf**
Fixes #1640 

Allows /z/:id/code/:sessionid to return plain text code.

I also tested if #include in c++ works, and it seems to work.

- [x] Cleanup / avoid code duplication
- [x] Add documentation

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Love it! Partouf you are - as ever - a legend :)

**Comment by partouf** [comment]
Changed link to /z/:shortlink/code/:sessionid

Refactored some of the code that was in app.js to a new file.

File: app.js [hunk 1]
```diff
@@ -43,10 +43,8 @@ const nopt = require('nopt'),
     {logger, logToPapertrail, suppressConsoleLog} = require('./lib/logger'),
     webpackDevMiddleware = require("webpack-dev-middleware"),
     utils = require('./lib/utils'),
-    clientState = require('./lib/clientstate'),
     initialiseWine = require('./lib/exec').initialiseWine,
-    clientStateGoldenifier = require('./lib/clientstate-normalizer').ClientStateGoldenifier,
-    clientStateNormalizer = require('./lib/clientstate-normalizer').ClientStateNormalizer;
+    RouteAPI = require('./lib/handlers/route-api');
```
Comment (mattgodbolt) [inline app.js]: Nice extraction of this: the more code we can remove from `app.js` the better!

File: docs/API.md [hunk 2]
```diff
@@ -235,6 +235,18 @@ Returns:
 
 The storedId can be used in the api call /api/shortlinkinfo/<id> and to open in the website with a /z/<id> shortlink.
 
+### `GET /z/<id>` - Opens the website from a shortlink
```
Comment (mattgodbolt) [inline docs/API.md]: In future I was thinking we might want to namespace `<id>`s like `GET /z/<username>/tag`. I'm pretty sure we can get this to work in either case (and/or change that idea...which is a long way off anyway! Was just thinking out loud)

File: package.json [hunk 3]
```diff
@@ -105,6 +105,7 @@
     "test": "mocha --recursive",
     "check": "npm run lint && npm run test",
     "dev": "cross-env NODE_ENV=DEV node app.js",
+    "debug": "cross-env NODE_ENV=DEV node app.js --debug",
```
Comment (mattgodbolt) [inline package.json]: Thanks :)

**Comment by partouf** [comment]
Tested on beta, don't see any issues

**Pull request merged**